packages/core/src/diff/changes/directive-usage.ts

```diff
@@ -0,0 +1,206 @@
+import { ConstDirectiveNode, GraphQLEnumType, GraphQLUnionType } from 'graphql';
+import {
+  Change,
+  ChangeType,
+  CriticalityLevel,
+  DirectiveUsageEnumAddedChange,
+  DirectiveUsageEnumRemovedChange,
+  DirectiveUsageEnumValueAddedChange,
+  DirectiveUsageEnumValueRemovedChange,
+  DirectiveUsageUnionMemberAddedChange,
+  DirectiveUsageUnionMemberRemovedChange,
+} from './change.js';
+
+export function directiveUsageAdded(
+  directive: ConstDirectiveNode,
+  path: string,
+): Change<ChangeType.DirectiveUsageAdded> {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+    },
+    message: `Directive '${directive.name.value}' was added`,
+    meta: {},
+    path,
+    type: ChangeType.DirectiveUsageAdded,
+  };
+}
+
+export function directiveUsageRemoved(
+  directive: ConstDirectiveNode,
+  path: string,
+): Change<ChangeType.DirectiveUsageRemoved> {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+    },
+    message: `Directive '${directive.name.value}' was removed`,
+    meta: {},
+    path,
+    type: ChangeType.DirectiveUsageRemoved,
+  };
+}
+
+export function directiveUsageBuildUnionMemberAddedMessageFromMeta(
+  args: DirectiveUsageUnionMemberAddedChange,
+) {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+      reason:
+        'Adding a union member to a union can cause existing queries that use this union member in a fragment spread to error.',
+    },
+    type: ChangeType.DirectiveUsageUnionMemberAdded,
+    message: `Directive '${args.meta.addedUnionMemberTypeName}' was added to Union type '${args.meta.unionName}'`,
```

Comment:
Not sure if it should be `added to Union` or `added on Union`. 
Also, I think it's better to just `union` and not `Union` here.